lib/bundler.rb

```diff
@@ -352,9 +352,9 @@ def load_gemspec_uncached(file, validate = false)
         spec
       end
     rescue Gem::InvalidSpecificationException => e
-      UI::Shell.new.warn "The gemspec at #{file} is not valid. " \
-        "The validation error was '#{e.message}'"
-      nil
+      UI::Shell.new.warn "The gemspec at #{file} is not valid. Please fix this gemspec.\n" \
+        "The validation error was '#{e.message}'\n"
+      raise Gem::InvalidSpecificationException
```

Comment:
So the reason that I essentially re-raise the error here is to wipe away the error message.

Otherwise, with raising the original error, the output is as follows:
```
       The gemspec at /Users/pivotal/workspace/bundler-rochester/tmp/libs/foo-1.0/foo.gemspec is not valid. Please fix this gemspec.
       The validation error was 'missing value for attribute version'
       You have one or more invalid gemspecs that need to be fixed.
       missing value for attribute version
```

instead of (with this new raised error):

```
       The gemspec at /Users/pivotal/workspace/bundler-rochester/tmp/libs/foo-1.0/foo.gemspec is not valid. Please fix this gemspec.
       The validation error was 'missing value for attribute version'
       You have one or more invalid gemspecs that need to be fixed.
       Gem::InvalidSpecificationException
```

The output might look even worse for multiple validation errors. Thoughts?